### PR TITLE
Add clearCoatNormal property

### PR DESCRIPTION
**Pull request by romainguy**
This new property can be used to set the normal map of the clear coat layer. This allows materials to have a different normal map for the base layer and the clear coat layer.

Each layer uses a different normal map

### CODE REVIEW

**Comment by romainguy** [comment]
This change also fixes a bug for the case when the base layer has a normal map but not the clear coat layer: the clear coat layer was multiplied by the base layer's `dot(N, V)` instead of the geometry's `dot(geometry_N, V)`.

**Comment by prideout** [comment]
From now on, all PR's must have beautiful images in the description.  :)

File: samples/sample_normal_map.cpp [hunk 1]
```diff
@@ -93,15 +97,16 @@ static void printUsage(char* name) {
...
-            { "scale",          required_argument, 0, 's' },
-            { "normal-map",     required_argument, 0, 'n' },
-            { "basecolor-map",  required_argument, 0, 'c' },
-            { 0, 0, 0, 0 }  // termination of the option list
+            { "help",                   no_argument,       nullptr, 'h' },
+            { "ibl",                    required_argument, nullptr, 'i' },
+            { "split-view",             no_argument,       nullptr, 'v' },
+            { "scale",                  required_argument, nullptr, 's' },
+            { "normal-map",             required_argument, nullptr, 'n' },
+            { "clear-coat-normal-map",  required_argument, nullptr, 'n' },
```
Comment (prideout) [inline samples/sample_normal_map.cpp]: Typo, replace `n` with `c`.

Comment (romainguy) [reply samples/sample_normal_map.cpp]: Great catch!

**Pull request merged**